components/payments/internal/app/api/router.go

```diff
@@ -31,10 +27,15 @@ func httpRouter(logger logging.Logger, store *storage.Storage, serviceInfo api.S
 	})
 
 	rootMux.Path("/_health").Handler(healthHandler(store))
-	rootMux.Path("/_live").Handler(liveHandler())
-	rootMux.Path("/_info").Handler(api.InfoHandler(serviceInfo))
 
-	authGroup := rootMux.Name("authenticated").Subrouter()
+	subRouter := rootMux.NewRoute().Subrouter()
+	if viper.GetBool(otelTracesFlag) {
+		subRouter.Use(otelmux.Middleware(serviceName))
```

Comment:
Put there, i think the middleware will no catch errors catched by the recoveryHandler. Can you check if otelmux catch panics?

Reply:
You're right, let me think about something to fix that !